discovery/consul_test.go

```diff
@@ -77,18 +77,18 @@ a Consul server for testing. The 'consul' binary must be in the $PATH
 ref https://github.com/hashicorp/consul/tree/master/testutil
 */
 
-var testServer *testutil.TestServer
+var testServer *TestServer
```

Comment:
Don't use global state for tests. Pass the test server or better just the http address into the sub tests since that is all they need AFAICT.

Reply:
I agree, artifact of the previous implementation.